tests/src/builders.rs

```diff
@@ -0,0 +1,139 @@
+//!#[doc(hidden)]
+
+/// # Doc tests
```

Comment:
Why do you use hidden doc tests?

Reply:
These are `compile_fail` test cases and other tests of generator behavior and corner cases that are of no interest to a normal library user.

Since the tests crate is `publish = false`, this should not matter much though.